src/core/shape_inference/include/multinomial_shape_inference.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "dimension_util.hpp"
+#include "openvino/op/multinomial.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v13 {
+template <class TShape, class TRShape = result_shape_t<TShape>>
+std::vector<TRShape> shape_infer(const Multinomial* op,
+                                 const std::vector<TShape>& input_shapes,
+                                 const ITensorAccessor& ta = make_tensor_accessor()) {
+    using TDim = typename TShape::value_type;
+
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 2);
+
+    const auto& input_shape = input_shapes[0];
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           input_shape.rank().compatible(1) || input_shape.rank().compatible(2),
+                           "The rank of the 'input' tensor defining output shape must be either 1 or 2.");
+
+    const auto& num_samples_shape = input_shapes[1];
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           num_samples_shape.compatible(TRShape{}) || num_samples_shape.compatible(TRShape{1}),
+                           "Number of samples must be a scalar or one element 1D tensor.");
+
+    auto output_shapes = std::vector<TRShape>(1);
+    auto& result_shape = output_shapes[0];
+    const auto input_shape_static = input_shape.is_static();
```

Comment:
```const auto input_shape_static = input_shape.is_static();```

This is very restriceted check as shape is static when rank and all dimension are static.
There is often perform some shape calculation when rank is static but dimensions can be interval values e.g 
`input_shape` is {{1,5}, 10}

Maybe this flag should hold if input_shape rank is static?

Reply:
Changed implementation to check for rank!